### ISSUE TITLE
Repository has bitrotted?

### ISSUE THREAD

**Issue by nmsmith** [nwf/dyna#87]
Has this repository bitrotted? I spent several hours just now trying to compile Dyna, but as someone who's not a Cabal expert, I was not able to get it to compile, no matter what version of Haskell or Cabal I tried. I kept getting package version clashes, primarily with wl-pprint-terminfo.

**Comment by nwf** [comment]
Yes, I imagine the repository has bitrotted quite substantially in the several years since I've completed my PhD; it was fragile even then.  If you want to play with a Dyna-like language, it's possible that @timvieira has better options available.

**Comment by nmsmith** [comment]
Yes I’d love to be able to play with a similar logic language, especially if there are no functional Dyna compilers! The language has piqued my interest.

**Comment by matthewfl** [comment]
We have a new iteration of dyna that we are working on here: https://github.com/matthewfl/dyna-R/ ([paper](http://cs.jhu.edu/~mfl/#Evaluation%20of%20Logic%20Programs%20with%20Built-Ins%20and%20Aggregation%3A%20A%20Calculus%20for%20Bag%20Relations:%5B%5BEvaluation%20of%20Logic%20Programs%20with%20Built-Ins%20and%20Aggregation%3A%20A%20Calculus%20for%20Bag%20Relations%5D%5D))

**Comment by nmsmith** [comment]
Exciting! I'll have a play around.

**Comment by nmsmith** [comment]
Also, it might be worth updating the README with this information to save the next poor soul some time :)

**Comment by PtrMan** [comment]
I tried to get a compiler which supports a extremely small subset of Dyna working https://github.com/PtrMan/Tenma
It currently has a compiler to Haxe and a untested one to OpenCL (I belief). This thing takes the AST of the Dyna program and statically compiles it to the target language(s).

This project died down, because I am currently not using Dyna in my projects and because I have other stuff to do. But I made sure that the compiler works for simple cases.

Maybe this thing can inspire some people to try something similar, good luck :)
